src/main/scala/view/scalaFX/components/WolfView.scala

```diff
@@ -45,39 +43,35 @@ object WolfView {
   }
 
   private case class WolfViewImpl(
+      factorsPanelController: Option[FactorsPanelControllerInterface],
       imageView: ImageView,
       var direction: Direction,
       var positionX: Double,
       var positionY: Double
   ) extends WolfView {
-    private val SPEED = 200
-    private val movingSpace = PREFERRED_SIMULATION_PANEL_WIDTH / SPEED
-    private val timer: AnimationTimer = AnimationTimer(t => {
-      if (lastTime > 0) {
-        checkDirection()
-        moveHorizontally()
+
+    private var lastTime = 0L
+
+    private val timer: AnimationTimer = AnimationTimer(_ => {
+      if (lastTime <= WOLVES_PHASE * 1000) {
+        checkDirection(
+          positionX + imageView.getFitWidth / 2 >= PREFERRED_SIMULATION_PANEL_WIDTH - PREFERRED_SIMULATION_PANEL_BORDER,
+          positionX - imageView.getFitWidth / 2 < 0
+        )
+        moveHorizontally(
+          WOLVES_MOVING_SPACE
+        )
         imageView.x = positionX
-      }
-      lastTime = t
+        lastTime += 1
+      } else stop()
     })
-    private var lastTime = 0L
 
     override def play(): Unit = timer.start()
```

Comment:
Per avere l'effetto dei lupi a cavallo della loro fase un'idea che mi è venuta è la seguente:  il baseAppController esegue la showWolves nella StartPhase ma in realtà i lupi vengono mostrati dopo un tot di millisecondi, cioè di base le loro imageView non sono visibili, e nell'AnimationTimer si aggiunge una condizione per rimetterle a visible dopo i millisecondi specificati

Reply:
Quindi se ho capito bene lasciamo il metodo per rimuovere i lupi alla fine della loro esecuzione, ma uno alla volta invece che tutti insieme, e li aggiungiamo subito ma si vedranno solo dopo un tot di tempo.
Perchè allora mi è venuto in mente che potremo aggiungerli una volta sola all'inizio e non rimuoverli mai ma semplicemente non visualizzarli più quando il loro periodo è finito oppure non sono stati selezionati dall'utente